### PR TITLE
Add Forseti template

### PR DESCRIPTION
**Pull request by sourced-pavel**

### CODE REVIEW

File: community/cloud-foundation/templates/forseti/client.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-""" Creates a GCE instance template for Forseti Security. """
+""" Creates a GCE instance template for Forseti Security.
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/forseti/client.py]: @alexander-levitski I'll add `client` and `server` words

File: community/cloud-foundation/templates/forseti/client.py.schema [hunk 2]
```diff
@@ -29,36 +29,38 @@ required:
...
   project:
     type: string
-    description: Project ID where the instance should be placed.
+    description: |
+      The project ID of the project where the instance must be placed.
   name:
     type: string
     default: forseti-client
-    description: The name of the Forseti server instance.
+    description: The name of the Forseti server ??? or client ??? instance.
```
Comment (sourced-pavel) [inline community/cloud-foundation/templates/forseti/client.py.schema]: Yup, client

File: community/cloud-foundation/templates/forseti/client.py (hunk 1 above)
Comment (alexander-levitski) [reply community/cloud-foundation/templates/forseti/client.py]: This template creates a GCE client for Forseti Security.
This template creates a GCE server for Forseti Security.

File: community/cloud-foundation/templates/forseti/client.py.schema (hunk 2 above)
Comment (alexander-levitski) [reply community/cloud-foundation/templates/forseti/client.py.schema]: So you'll change it? Or should I?

Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/client.py.schema]: I changed that already

File: community/cloud-foundation/templates/forseti/forseti.py.schema [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+      zone:
+        type: string
+        description: The availability zone. E.g., 'us-central1-a'.
+      port:
+        type: number
+        default: 3306
+        description: The Cloud SQL port number.
+      frequency:
+        type: string
+        default: '0 */2 * * *'
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/forseti.py.schema]: It's every other hour. CRON [supports](https://crontab.guru/every-2-hours) that

File: community/cloud-foundation/templates/forseti/forseti.py.schema [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+          The Compute Instance type; e.g., 'n1-standard-1'.
+          See https://cloud.google.com/compute/docs/machine-types for details.
+      sqlOsArch:
+        type: string
+        default: linux.amd64
+        description: The architecture of the Cloud SQL proxy to use.
+  client:
+    type: object
+    description: The Forseti client instance configuration.
+    required:
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/forseti.py.schema]: They are different - forseti.py takes care of a number of parameters for client and a server, so `client` in `forseti.py.schema` is smaller than in `client.py.schema`

File: community/cloud-foundation/templates/project/project.py [hunk 5]
```diff
@@ -82,6 +82,12 @@ def generate_config(context):
                         'serviceAccountDisplayName',
                     'value':
                         '$(ref.project.projectNumber)@cloudservices.gserviceaccount.com'  # pylint: disable=line-too-long
+                },
+                {
+                    'name':
+                        'resources',
+                    'value':
+                        [resource['name'] for resource in resources]
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/project/project.py]: The purpose of this output is to be a sync context for other, 'consequent' resources. We need the way to wait for everything to finish in the project template, before creating anything else. `resources` output is meant to be passed to `dependsOn` metadata.
We used the same principle in `cloudsql`

File: community/cloud-foundation/templates/forseti/client.py.schema [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+info:
+  title: Forseti Security
+  author: Sourced Group Inc.
+  description: |
+    Supports creation of a Forseti Security client instance.
+
+required:
+  - project
+  - zone
+  - bucket
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/client.py.schema]: Fixed

File: community/cloud-foundation/templates/forseti/forseti.py.schema [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+      deployed.
+  server:
+    type: object
+    description: The Forseti server config.
+    required:
+      - zone
+    properties:
+      serviceAccount:
+        type: object
+        description: |
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/forseti.py.schema]: Fixed

File: community/cloud-foundation/templates/forseti/forseti.py.schema [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+          Forseti server's new service account settings.
+        properties:
+          accountId:
+            type: string
+            description: the optional ID of the service account.
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/forseti.py.schema]: Fixed

File: community/cloud-foundation/templates/forseti/server.py [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+# If the cron job failed the acquire lock on the process, it will log a warning message to syslog.
+(echo "{run_frequency} (/usr/bin/flock -n /home/ubuntu/forseti-security/forseti_cron_runner.lock $FORSETI_HOME/install/gcp/scripts/run_forseti.sh || echo '[forseti-security] Warning: New Forseti cron job will not be started, because previous Forseti job is still running.') 2>&1 | logger") | crontab -u $USER -
+echo "Added the run_forseti.sh to crontab under user $USER"
+echo "Execution of startup script finished"
+"""
+
+def get_export_initialize_vars(database, port, connection_string):
+    """ Gets the shell script that persists the Forseti env variables. """
+
+    template = (
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/server.py]: Fixed

File: community/cloud-foundation/tests/cloud-foundation-tests.conf.example [hunk 10]
```diff
@@ -1,5 +1,6 @@
 # Please change the values for these variables to match your organization's GCP setup
 export CLOUD_FOUNDATION_ORGANIZATION_ID="111111111111"
 export CLOUD_FOUNDATION_PROJECT_ID="my-project"
+export CLOUD_FOUNDATION_FOLDER_ID="000000000000"
```
[the remark this answers is not in the document]
Comment (sourced-pavel) [reply community/cloud-foundation/tests/cloud-foundation-tests.conf.example]: Fixed

File: community/cloud-foundation/templates/forseti/tests/integration/forseti.bats [hunk 11]
```diff
@@ -49,8 +50,18 @@ function setup() {
...
+        gcloud alpha resource-manager folders create \
+            --display-name="${CLOUD_FOUNDATION_FOLDER_NAME}" \
+            --organization="${CLOUD_FOUNDATION_ORGANIZATION_ID}" > ~/output.txt
+        export CLOUD_FOUNDATION_FOLDER_ID=$(gcloud alpha resource-manager folders list \
+            --project "${CLOUD_FOUNDATION_PROJECT_ID}" \
+            --organization "${CLOUD_FOUNDATION_ORGANIZATION_ID}" | \
+            grep "${CLOUD_FOUNDATION_FOLDER_NAME}" | \
+            awk '{print $3}')
+        gcloud alpha resource-manager folders list --organization="${CLOUD_FOUNDATION_ORGANIZATION_ID}" >> ~/output.txt
+        echo "${CLOUD_FOUNDATION_FOLDER_ID}" >> ~/output.txt
```
Comment (sourced-vince) [inline community/cloud-foundation/templates/forseti/tests/integration/forseti.bats]: Remove

File: community/cloud-foundation/templates/forseti/tests/integration/forseti.bats [hunk 12]
```diff
@@ -72,6 +85,7 @@ function teardown() {
     run gcloud deployment-manager deployments create "${DEPLOYMENT_NAME}" \
         --config "${CONFIG}" \
         --project "${CLOUD_FOUNDATION_PROJECT_ID}"
+    echo "$output" >> ~/output.txt
```
Comment (sourced-vince) [inline community/cloud-foundation/templates/forseti/tests/integration/forseti.bats]: Remove

File: community/cloud-foundation/templates/forseti/tests/integration/forseti.bats (hunk 11 above)
Comment (sourced-pavel) [reply community/cloud-foundation/templates/forseti/tests/integration/forseti.bats]: Oh, shi... Sorry, this is embarrassing.

**Pull request closed without merging**